src/main/java/com/intuit/graphql/authorization/enforcement/RedactingVisitor.java

```diff
@@ -53,6 +59,33 @@ public void visitField(QueryVisitorFieldEnvironment queryVisitorFieldEnvironment
       instrumentationState.getAuthzErrors().add(error);
 
       TreeTransformerUtil.deleteNode(queryVisitorFieldEnvironment.getTraverserContext());
+    } else {
```

Comment:
I think this a a good design, we should standardize it to make it cleaner: 
* Offline access should also follow the same pattern i.e. It should implement AuthorizationExtension. 
* It should support a list of `AuthorizationExtension` instances. Access is authorized only if all extensions permit access to a field.
* Offline access implementation could live in this library as a default. Clients can choose to append or override the list.